tools/src/wyvern/tools/ArchitectureInterpreter.java

```diff
@@ -0,0 +1,321 @@
+                if (seq instanceof SeqExpr) {
+                    for (HasLocation elem : ((SeqExpr) seq).getElements()) {
+                        if (elem instanceof VarBinding && ((VarBinding) elem).getVarName().contains(comp.getName())) {
+                            orderedExprs.add(seq);
+                            break;
+                        }
+                    }
+                } else {
+                    System.out.println("error?");
+                }
+            }
+        }
+        return orderedExprs;
+    }
+
+    private static DeclCheckVisitor checkArchFile(String rootLoc, String wyvernPath, String filename,
+                                                  Path filepath, InterpreterState state) throws ParseException, FileNotFoundException {
+        File f = new File(rootLoc + "/" + filepath.toString());
+        BufferedReader source = new BufferedReader(new FileReader(f));
+        ArchParser wp = new ArchParser(
+                (TokenManager) new WyvernTokenManager<ArchLexer, ArchParserConstants>(
+                        source, "test", ArchLexer.class, ArchParserConstants.class));
+        wp.fname = filename;
+        Node start = wp.ArchDesc();
+        DeclCheckVisitor visitor = new DeclCheckVisitor(state);
+        visitor.visit((ASTArchDesc) start, null);
+        return visitor;
+    }
+
+    private static ArrayList<Expression> unwrapGeneratedAST(int numPortAST, Value connectorImpl, InterpreterState state, EvalContext evalCtx) {
+        ArrayList<Expression> portInstances = new ArrayList<>();
+        Value getFirst = ((Invokable) connectorImpl).invoke("_getFirst", new LinkedList<>()).executeIfThunk();
+        // getFirst is an option
+        Value value = ((Invokable) getFirst).getField("value");
+        // value is an internal.list
+        for (int i = 0; i < numPortAST; i++) {
+            List<Value> invokeArgs = new LinkedList<>();
+            invokeArgs.add(new IntegerLiteral(i));
+            Value fromIList = ((Invokable) value).invoke("get", invokeArgs).executeIfThunk();
+            // fromIList is another other
+            Value option2 = ((Invokable) fromIList).getField("value");
+            // a wyvern AST !!!
+            JavaValue ast = (JavaValue) ((Invokable) option2).getField("ast");
+            // a JavaValue !!!!!!!!!!!!!!!!
+            JObject obj = (JObject) ast.getFObject();
+            Object javaAST = obj.getWrappedValue();
+            if (javaAST instanceof New) {
+                // Get module def ASTs from generateConnectorImpl
+                Expression newAST = (New) javaAST;
+                for (Declaration decl : ((New) newAST).getDecls()) {
+                    if (decl instanceof ModuleDeclaration) {
+                        String moduleName = ((ModuleDeclaration) decl).getName();
+                        // state.addModuleAST(moduleName, newAST);
+                        // System.out.println(moduleName);
+                    }
+                }
```

Comment:
What is this loop for? It seems to me that it is only here for debugging. If that is true, please comment it out.